### ISSUE TITLE
Allow "extending" of a certificate to add new domains.

### ISSUE THREAD

**Issue by mithro** [kuba/simp_le#36]
I tried to add another vhost to an existing certificate and got the following error;

```
Traceback (most recent call last):
  File "/home/letsencrypt/simp_le/venv/bin/simp_le", line 9, in <module>
    load_entry_point('simp-le', 'console_scripts', 'simp_le')()
  File "/home/letsencrypt/simp_le/simp_le.py", line 871, in main
    raise SystemExit(_main(cli_args))
  File "/home/letsencrypt/simp_le/simp_le.py", line 859, in _main
    if _valid_existing_data(args.ioplugins, args.vhosts, args.valid_min):
  File "/home/letsencrypt/simp_le/simp_le.py", line 746, in _valid_existing_data
    assert set(existing_sans) == set(vhost.name for vhost in vhosts)
AssertionError
```

Following patch seems to make things work okay;

``` patch
diff --git a/simp_le.py b/simp_le.py
index c827b16..a9c7bab 100755
--- a/simp_le.py
+++ b/simp_le.py
@@ -741,9 +741,12 @@ def _valid_existing_data(ioplugins, vhosts, valid_min):
     if existing != IOPlugin.EMPTY_DATA:
         # pylint: disable=protected-access
         existing_sans = crypto_util._pyopenssl_cert_or_req_san(existing.cert)
+        requested_sans = set(vhost.name for vhost in vhosts)
         logger.debug('Existing SANs: %r', existing_sans)

-        assert set(existing_sans) == set(vhost.name for vhost in vhosts)
+        if set(existing_sans) != requested_sans:
+            logger.debug('New SANs: %r', requested_sans)
+            return False

         # Renew?
         if not renewal_necessary(existing.cert, valid_min):
```

**Comment by Lekensteyn** [comment]
Duplicate of #21

**Comment by ThomasWaldmann** [comment]
Well, it is somewhat duplicate, but adding a name should not be considered a fault. E.g. you forgot www. in first try and now you want to add it. Such use cases are expected.

**Comment by kuba** [comment]
What if you misspelled the command and forgot to list a name that you used before? I'd rather not make any magical steps to try to fix it and prefer the user to take appropriate manual action, especially because this software overwrites previously stored certs - if user didn't have backups then they might have just lost their cert and they might need to wait until rate limit is reset. I've added necessary error messages in c78aaec and 027a5d2.

**Comment by ThomasWaldmann** [comment]
"Backup and remove existing persisted data if you want to proceed."

Maybe that should be more specific. Don't assume everyone knows how it internally works.

**Comment by mithro** [comment]
Don't you want to keep the private key?

**Comment by kuba** [comment]
FTR As of https://github.com/kuba/simp_le/commit/09026580ece34227940e3e4b16fd603397b867ce this messages is now: "Backup and remove existing cert if you want to proceed."

**Comment by emilyalbini** [comment]
Could you add an `--allow_overwrite` flag to allow overwriting? I have a script which generates and executes simp_le with the domains taken from a file, so there is no "I forgot a domain" problem, and ansible adds domains to that file when a new thing is hosted on the machine.

With that flag I could simply re-execute the script to get the right cert, instead of deleting certificates ahead causing some issues with the services on the machine. I can provide a patch if you want.

**Comment by kuba** [comment]
That was fixed in  e30ad8c.
